mod_app/models/support_models.py

```diff
@@ -22,96 +25,74 @@ def __str__(self):
+        verbose_name = "Source"
+
+    is_source = models.BooleanField(default=True)
+
+
 class FileLink(Link):
     def upload_to(instance, filename):
-        return f"files/{instance.related_name}/{filename}"
+        return f"files/{instance.__class__.__name__}/{filename}"
 
     file = models.FileField(upload_to=upload_to, blank=True, null=True)
 
 
-class Copy(Link):
+class Script(FileLink):
     class Meta:
-        verbose_name_plural = "Copies"
+        verbose_name = "Script"
 
-    def __str__(self):
-        if self.name:
-            return self.name
-        else:
-            return super().__str__()
 
-    name = models.CharField(
-        max_length=100,
-        help_text="optional name/title for the copy",
-        blank=True,
-    )
+class PressBook(FileLink):
+    class Meta:
+        verbose_name = "Press Book"
 
-    CONDITION_CHOICES = [
-        ("good", "Good"),
-        ("reasonable", "Reasonable"),
-        ("poor", "Poor"),
-        ("fragile", "Fragile"),
-        ("other", "Other"),
-    ]
-
-    condition = models.CharField(
-        max_length=10,
-        choices=CONDITION_CHOICES,
-        default="reasonable",
-    )
 
-    condition_comments = models.TextField(
-        help_text="Use this to expand on the condition, particularly if 'Other'",
-        blank=True,
-    )
+class Programme(FileLink):
+    class Meta:
+        verbose_name = "Programme"
 
 
-class Analysis(models.Model):
+class Publicity(FileLink):
     class Meta:
-        verbose_name_plural = "Analyses"
+        verbose_name = "Publicity Material"
 
-    def __str__(self):
-        return f"Analysis of {self.film}"
 
-    content = RichTextField(null=True)
-    film = models.ForeignKey(
-        "Film", on_delete=models.DO_NOTHING, related_name="analyses"
-    )
+class Still(FileLink):
+    class Meta:
+        verbose_name = "Still"
 
 
-class Actor(models.Model):
-    def __str__(self):
-        return f"{self.name}"
+class Postcard(FileLink):
+    class Meta:
+        verbose_name = "Postcard"
 
-    name = models.CharField(max_length=100)
-    birth_place = models.ForeignKey(
-        "Location", on_delete=models.SET_DEFAULT, default=None, blank=True, null=True
-    )
-    tags = models.ManyToManyField(Tag, related_name="actor_tags", blank=True)
-    about = models.TextField(blank=True)
+
+class Poster(FileLink):
+    class Meta:
+        verbose_name = "Poster"
+
+
+class Drawing(FileLink):
+    class Meta:
+        verbose_name = "Drawing"
 
 
-class Crew(models.Model):
+class Analysis(models.Model):
     class Meta:
-        verbose_name_plural = "Crew Members"
+        verbose_name_plural = "Analyses"
 
     def __str__(self):
-        return f"{self.name}"
+        return f"Analysis of {self.film}"
 
-    ROLE_CHOICES = [
-        ("Director", "Director"),
-        ("Screenplay", "Screenplay"),
-        ("Scenography", "Scenography"),
-        ("Photography", "Photography"),
-    ]
+    content = RichTextField(null=True)
```

Comment:
I'm curious what RichTextField adds that TextField does have? Why use it here and not on Film.synopsis or comments?

Reply:
I've done it in different ways because on the film admin I can override all textfields at once to use the ckeditor, and in models where there's only like one textfield I've done it through the models and then utilise a slightly different way of enabling the widget for the admin (you have to declare for each field, which is fine when there's only 1, but a nuisance when there's 6)